project-koku/koku
# Cost 2138 HCS pipeline for AWS

Fixes 
    [COST-2138](https://issues.redhat.com/browse/COST-2138)
    [COST-2139](https://issues.redhat.com/browse/COST-2139)

Test instructions:    
1. Setup dev env
        ` make create-test-customer`
         `make load-test-customer-data`    
  
2. Watch the logs in one terminal by running the following command
         `docker logs koku_koku-worker_1 -f`   
   
3. Run the following masu end point
         http://localhost:5042/api/cost-management/v1)/hcs_report_data/?schema=10001&provider_type=AWS&provider_uuid=eafd727f-ed1c-4c14-aae9-0e0220db8af3&start_date=20220327  
     
4. If you have not added the account to the Unleash server you should see the following in the log output

> [2022-03-28 20:58:44,692: INFO/ForkPoolWorker-3] enable_hcs_processing context: {'schema': 'accteafd727f-ed1c-4c14-aae9-0e0220db8af3'}
[2022-03-28 20:58:44,693: INFO/ForkPoolWorker-3] {'message': '[SKIPPED] Customer not registered with HCS: Schema-name: 10001, provider: AWS, provider_uuid: eafd727f-ed1c-4c14-aae9-0e0220db8af3, dates 20220322 - 20220328', 'tracing_id': 'a16ca8f1-9b68-4bc2-9bf5-8207abdc0f8c'}
     
5. Add `acct10001` to unleash server
     `http://localhost:4242/`
     login
     and select `hcs-data-processor`
     add acct10001 to List of schema-name (see below image)
     
![Screen Shot 2022-03-29 at 7 20 55 AM](https://user-images.githubusercontent.com/57504257/160600447-29ffb458-d431-47ce-86a9-351e68777289.png)

6. Run the same masu end point(from step 3). Now you should see something like the below in the logs

> 2022-03-29 02:02:23,725: INFO/ForkPoolWorker-6] enable_hcs_processing context: {'schema': 'acct10001'}
[2022-03-29 02:02:23,725: INFO/ForkPoolWorker-6] {'message': 'Running HCS data collection: schema_name: acct10001, provider_uuid: eafd727f-ed1c-4c14-aae9-0e0220db8af3, provider: AWS, dates 20220327 - 20220329', 'tracing_id': '5217ba54-fc3c-4fc9-8be1-85ed01f1375b'}
[2022-03-29 02:02:23,736: INFO/ForkPoolWorker-6] {'message': 'acquiring marketplace data...', 'tracing_id': '5217ba54-fc3c-4fc9-8be1-85ed01f1375b'}
[2022-03-29 02:02:23,736: INFO/ForkPoolWorker-6] {'message': 'schema: acct10001,  provider: AWS, date: 2022-03-27', 'tracing_id': '5217ba54-fc3c-4fc9-8be1-85ed01f1375b'}
[2022-03-29 02:02:24,546: INFO/ForkPoolWorker-6] {'message': 'data found for date: 2022-03-27', 'tracing_id': '5217ba54-fc3c-4fc9-8be1-85ed01f1375b'}
[2022-03-29 02:02:24,564: INFO/ForkPoolWorker-6] {'message': 'preparing to write file to object storage', 'tracing_id': '5217ba54-fc3c-4fc9-8be1-85ed01f1375b'}

7. look to see that the data has been pushed to object storage (using MinIO)
http://localhost:9090/
Look under the kolu bucket you should see a `hcs` (see below)

![Screen Shot 2022-03-30 at 9 35 42 AM](https://user-images.githubusercontent.com/57504257/160847388-e007d5e5-3d94-4f7f-826a-3635bcdf716c.png)
